qiskit/primitives/base_estimator.py

```diff
@@ -141,8 +147,11 @@ def __init__(
         Raises:
             QiskitError: For mismatch of circuits and parameters list.
         """
-        self._circuits = tuple(circuits)
-        self._observables = tuple(observables)
+        self._circuits = circuits
+        self._observables = observables
+        self._circuit_names = self._circuit_names
+        self._observable_ids = self._observable_ids
```

Comment:
It might be nice to add a comment explaining these lines, otherwise it looks a bit redundant like Imamichi-san mentioned above. 

But aside from that I still don't quite understand why we need to go via `__new__` for this and can't just put this in `__init__` -- all subclasses should call `super().__init__` anyways, shouldn't they? 🤔

Reply:
Thank you. I will add such a comment.
The use of `__new__` is a matter of processing order. The object at `super().__init__` may have been changed, so the names may have changed or the IDs may have changed.